xds/src/test/java/io/grpc/xds/ClientXdsClientDataTest.java

```diff
@@ -773,245 +803,197 @@ public void parseCluster_ringHashLbPolicy_invalidRingSizeConfig_tooLargeRingSize
             .addFilterChains(filterChain)
+            .setAddress(
+                Address.newBuilder()
+                    .setSocketAddress(
+                        SocketAddress.newBuilder().setAddress("10.0.1.2").setPortValue(8080)))
             .build();
-    StructOrError<io.grpc.xds.EnvoyServerProtoData.Listener> struct =
-        ClientXdsClient.parseServerSideListener(listener, null);
-    assertThat(struct.getErrorDetail())
-        .isEqualTo(
-            "filter envoy.http_connection_manager with unsupported typed_config type:badTypeUrl");
+    thrown.expect(ResourceInvalidException.class);
+    thrown.expectMessage(
+        "FilterChain filter-chain-foo contains filter envoy.http_connection_manager with "
+            + "unsupported typed_config type badTypeUrl");
+    ClientXdsClient.parseServerSideListener(listener, null, true);
   }
 
   @Test
-  public void parseServerSideListener_duplicateHttpFilter() {
+  public void parseServerSideListener_duplicateHttpFilter() throws ResourceInvalidException {
     Filter filter =
-        buildHttpConnectionManager(
-            "envoy.http_connection_manager",
+        buildHttpConnectionManagerFilter(
             HttpFilter.newBuilder().setName("hf").setIsOptional(true).build(),
             HttpFilter.newBuilder().setName("hf").setIsOptional(true).build());
-    FilterChain filterChain = buildFilterChain(filter);
-    Listener listener =
-        Listener.newBuilder()
-            .setName("listener1")
-            .setTrafficDirection(TrafficDirection.INBOUND)
-            .addFilterChains(filterChain)
-            .build();
-    StructOrError<io.grpc.xds.EnvoyServerProtoData.Listener> struct =
-        ClientXdsClient.parseServerSideListener(listener, null);
-    assertThat(struct.getErrorDetail())
-        .isEqualTo("http-connection-manager has non-unique http-filter name:hf");
-  }
-
-  @Test
-  public void parseServerSideListener_configDiscoveryHttpFilter() {
-    Filter filter =
-        buildHttpConnectionManager(
-            "envoy.http_connection_manager",
-            HttpFilter.newBuilder()
-                .setName("envoy.router")
-                .setConfigDiscovery(ExtensionConfigSource.newBuilder().build())
-                .build());
-    FilterChain filterChain = buildFilterChain(filter);
-    Listener listener =
-        Listener.newBuilder()
-            .setName("listener1")
-            .setTrafficDirection(TrafficDirection.INBOUND)
-            .addFilterChains(filterChain)
-            .build();
-    StructOrError<io.grpc.xds.EnvoyServerProtoData.Listener> struct =
-        ClientXdsClient.parseServerSideListener(listener, null);
-    assertThat(struct.getErrorDetail())
-        .isEqualTo(
-            "http-connection-manager http-filter envoy.router uses "
-                + "config-discovery which is unsupported");
-  }
-
-  @Test
-  public void parseServerSideListener_badTypeUrlHttpFilter() {
```

Comment:
Is this also being tested elsewhere?

Reply:
See `parseHttpFilter_unsupportedButOptional()`, `parseHttpFilter_unsupportedAndRequired()`